pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -159,6 +162,15 @@ func (t ResolvedPipelineRunTask) IsSkipped(state PipelineRunState, d *dag.Graph)
 		}
 	}
 
+	// Check if the when expressions are false, based on the input's relationship to the values
+	if len(t.PipelineTask.WhenExpressions) > 0 {
+		if !t.PipelineTask.WhenExpressions.HaveVariables() {
```

Comment:
What is the significance of `HaveVariables` here?

Reply:
validating that we're not determining that a task is skipped based on an unreplaced variable in a when expression